src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -86,33 +91,40 @@ public void start_backup()
         /// </remarks>
         public void reset_config(bool removeBackup = false)
         {
-            if (_originalConfiguration == null)
+            if (_configurationBackups == null || _configurationBackups.Count == 0)
             {
                 if (removeBackup)
                 {
                     // If we will also be removing the backup, we do not care if it is already
                     // null or not, as that is the intended state when this method returns.
+                    this.Log().Debug("Requested removal of a configuration backup that does not exist: the backup stack is empty.");
                     return;
                 }
 
                 throw new InvalidOperationException("No backup has been created before trying to reset the current configuration, and removal of the backup was not requested.");
             }
 
-            var t = this.GetType();
+            var t = typeof(ChocolateyConfiguration);

```

Comment:
This won't work as expected, remember that the type will be different in case of choco-licensed.

We need to do `this.GetType()` to get the actual type.

Reply:
I'll revert this and add a comment to this effect, thanks!

Reply:
Fixed!